### ISSUE TITLE
Cannot append java.time.LocalDateTime to DateTimeColumn

### ISSUE THREAD

**Issue by maratermeshev** [jtablesaw/tablesaw#1161]
**Happens here:**
https://github.com/jtablesaw/tablesaw/blob/c6634bf63e0e19fa54472c25702ac08230d19537/core/src/main/java/tech/tablesaw/api/InstantColumn.java#L276

**Called by:**
https://github.com/jtablesaw/tablesaw/blob/66749074c11fa6b81d9516bd49eb3f3798ea4c33/core/src/main/java/tech/tablesaw/io/jdbc/SqlResultSetReader.java#L127

**Steps to reproduce:**
1. Prepare data.
```
CREATE TABLE test_date_time
(
    id    int(11) UNSIGNED NOT NULL AUTO_INCREMENT,
    date_time DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
    PRIMARY KEY (`id`)
);
INSERT INTO test_date_time () VALUES ();
```

2. Follow the instruction here, the "Importing from a Database" section: https://jtablesaw.github.io/tablesaw/userguide/importing_data.html
Replace 'String sql = "SELECT * FROM Customer";' with 'String sql = "select * from test_date_time;";'

**Analyzing:**
If you put a breakpoint here: https://github.com/jtablesaw/tablesaw/blob/66749074c11fa6b81d9516bd49eb3f3798ea4c33/core/src/main/java/tech/tablesaw/io/jdbc/SqlResultSetReader.java#L127

and then evaluate:
metaData.getColumnLabel(2)
// date_time
metaData.getColumnType(2)
// 93

According to this: https://github.com/corretto/corretto-8/blob/9109be14f97cddcbbac2bf2cae391caeb924849a/jdk/src/share/classes/java/sql/Types.java#L148

93 stands for TIMESTAMP

Let's return to TableSaw:
https://github.com/jtablesaw/tablesaw/blob/66749074c11fa6b81d9516bd49eb3f3798ea4c33/core/src/main/java/tech/tablesaw/io/jdbc/SqlResultSetReader.java#L56

This is how MySQL DATETIME column type becomes Instant in TableSaw.

Please, put a breakpoint here:
https://github.com/jtablesaw/tablesaw/blob/c6634bf63e0e19fa54472c25702ac08230d19537/core/src/main/java/tech/tablesaw/api/InstantColumn.java#L276

and evaluate:
obj.getClass().getName()
// java.time.LocalDateTime

This is confusing because MySQL DATETIME lack time zone and JDBS converts it to LocalDateTime which has time zone. Anyway, I was wondering if it is possible to add additional check for LocalDateTime and to try to convert it to Instant?

**Comment by lwhite1** [comment]
LocalDateTime does not have a timezone. That would be a ZonedDateTime

**Comment by maratermeshev** [comment]
@lwhite1 
Do you think that it is a good idea to additional check for LocalDateTime to InstantColumn's appendOb method?
I can submit a pull request if you like the idea

**Comment by lwhite1** [comment]
I don't think you can auto-convert LocalDateTime to Instant in a meaningful
way. An Instant is supposed to be a unique point on the timeline, but a
LocalDateTime could be many points, depending on the (unspecified) time
zone.

**Comment by maratermeshev** [comment]
@lwhite1 I was wondering if you have any idea how to fix this issue?

**Comment by lwhite1** [comment]
Hi,

Could you please write a couple of clear sentences saying what you want to
happen and what is happening? It's hard to understand your intent from the
links and example code. i'm not set up to create a MySQL database and
execute SQL against it to debug issues.

**Comment by maratermeshev** [comment]
@lwhite1 I'm sorry that I wasn't clear.

1. I want this to work without errors:
```Table t = Table.read().db(ResultSet resultSet, String tableName);```
I don't want to meet any exceptions here.
2. This exception thrown:
```Cannot append java.time.LocalDateTime to DateTimeColumn```
It happens because of field with DATETIME type in the MySQL 5.7 database.
TableSaw is trying to cast LocalDateTime to InstantColumn
In this function:
```
  /** {@inheritDoc} */
  @Override
  public InstantColumn appendObj(Object obj) {
    if (obj == null) {
      return appendMissing();
    }
    if (obj instanceof Instant) {
      return append((Instant) obj);
    }
    if (obj instanceof Timestamp) {
      Timestamp timestamp = (Timestamp) obj;
      return append(timestamp.toInstant());
    }
    throw new IllegalArgumentException(
        "Cannot append " + obj.getClass().getName() + " to DateTimeColumn");
  }
```
But it can work only with Instant and Timestamp. It can't work with LocalDateTime because there is no code for LocalDateTime in this function.

So how is it happening that TableSaw is trying to cast LocalDateTime to InstantColumn while the dedicated DateTimeColumn is there for LocalDateTime type?

https://github.com/jtablesaw/tablesaw/blob/c6634bf63e0e19fa54472c25702ac08230d19537/core/src/main/java/tech/tablesaw/api/DateTimeColumn.java#L271

The answer is in the Java JDK, it returns 93 for DATETIME type, which Tablesaw interprets as Instant.

Conclusions:
1. MySQL DATETIME type falls into InstantColumn instead of dedicated DateTimeColumn.
2. MySQL DATETIME type turns into LocalDateTime.
3. InstantColumn can't parse LocalDateTime, and it causes an exception. 
How to fix it?

**Comment by lwhite1** [comment]
There seems to be an issue with the SQL type mapping logic.

First thing is the error message is misleading due to a cut and paste
error. It should obviously say

```
throw new IllegalArgumentException(

        "Cannot append " + obj.getClass().getName() + " to InstantColumn");

```

I'm not a SQL expert, but it seems that when you declare the SQL type
as DATETIME, JDBC reports the SQL type as TIMESTAMP (apparently
DATETIME is considered a synonym for TIMESTAMP), so Tablesaw creates
an InstantColumn to hold the result since Timestamps may have
timezones. A SQL TIMESTAMP may or may not have one. This is the root
of the problem.

I could add an if statement for LocalDateTime in the InstantColumn
class, but to convert that to an Instant, you have to provide a
timezone argument:

```

Instant instant = localDateTime.toInstant(ZoneOffset.UTC);

```

I can't be sure what Timezone to use. I hesitate to assume UTC, as
that could hide a programming bug in the client code that meant that
the data was mapped to the wrong zone and the analysis results were
thus incorrect, but I think that's the only way to handle this case.

In the short term, maybe consider splitting the column into two parts:
a DATE part and a TIME part, which isn't great I know.

**Comment by maratermeshev** [comment]
@lwhite1 
MySQL stores DATETIME without timezone information, so for the case that I caught this would do:
```
Instant instant = localDateTime.toInstant(ZoneOffset.UTC);
```

**Comment by lwhite1** [comment]
@maratermeshev Yes, that's the fix. In the meantime, you might be able to workaround as suggested, or maybe by providing a timezone when you put data in the table.

**Comment by maratermeshev** [comment]
@lwhite1 according to this: https://dev.mysql.com/doc/refman/5.7/en/datetime.html
```
 The DATETIME type is used for values that contain both date and time parts. MySQL retrieves and displays DATETIME values in 'YYYY-MM-DD hh:mm:ss' format. The supported range is '1000-01-01 00:00:00' to '9999-12-31 23:59:59'.

The TIMESTAMP data type is used for values that contain both date and time parts. TIMESTAMP has a range of '1970-01-01 00:00:01' UTC to '2038-01-19 03:14:07' UTC. 
```

Conclusion:
I didn't mention TIMESTAMP, the issue is related to DATETIME. TIMESTAMP has time zone, while DATETIME doesn't have time zone.
